### PR TITLE
[Product] Fix bug related with pricing in product show page

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | 1.6
| Bug fix?        |yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Screenshot 2019-10-01 at 09 40 18

replaced by https://github.com/Sylius/Sylius/pull/10999

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig [hunk 1]
```diff
@@ -11,14 +11,15 @@
                 </tr>
             </thead>
             <tbody>
-            {% for channelPricing in product.variants.first.channelPricings %}
+                {% set currencies = sylius_channels_currencies() %}
+                {% for channelPricing in product.variants.first.channelPricings %}
+                {% set channel = channelPricing.channelCode|sylius_channel %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig]: Wrong indentation

File: src/Sylius/Bundle/AdminBundle/Twig/ChannelExtension.php [hunk 2]
```diff
@@ -14,10 +14,11 @@
 namespace Sylius\Bundle\AdminBundle\Twig;
 
 use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
 use Twig\Extension\AbstractExtension;
 use Twig\TwigFilter;
 
-final class ChannelNameExtension extends AbstractExtension
+final class ChannelExtension extends AbstractExtension
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Twig/ChannelExtension.php]: I guess that this change could be a BC break

File: src/Sylius/Bundle/AdminBundle/Twig/ChannelExtension.php [hunk 3]
```diff
@@ -30,12 +31,12 @@ public function __construct(ChannelRepositoryInterface $channelRepository)
     public function getFilters(): array
     {
         return [
-            new TwigFilter('sylius_channel_name', [$this, 'getChannelNameByCode']),
+            new TwigFilter('sylius_channel', [$this, 'getChannelByCode']),
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Twig/ChannelExtension.php]: I'm not sure if it is not a BC break, I would be in favor of adding a new extension with this twig filter

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig [hunk 4]
```diff
@@ -1,23 +1,21 @@
...
         <tr>
             <td class="gray text"><strong>{{ 'sylius.ui.channels'|trans }}</strong></td>
             <td class="gray text"><strong>{{ 'sylius.ui.price'|trans }}</strong></td>
             <td class="gray text"><strong>{{ 'sylius.ui.original_price'|trans }}</strong></td>
         </tr>
-        {% set currencies = sylius_channels_currencies() %}
-        {% for channelPricing in variant.channelPricings %}
+            {% set currencies = sylius_channels_currencies() %}
+            {% for channelPricing in variant.channelPricings %}
+            {% set channel = channelPricing.channelCode|sylius_channel %}
```
Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig]: Should be indented because of the loop.

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig [hunk 5]
```diff
@@ -1,23 +1,21 @@
 {% import "@SyliusAdmin/Common/Macro/money.html.twig" as money %}
 
-<div class="ui segment">
-    <div class="ui small header">
-        {{ 'sylius.ui.pricing'|trans }}
-    </div>
+    <h4 class="ui top attached large header">{{ 'sylius.ui.pricing'|trans }}</h4>
```
Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig]: Why it is indented?

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig [hunk 6]
```diff
@@ -11,19 +11,20 @@
...
-                        <td>-</td>
-                    {% endif %}
-                </tr>
-            {% endfor %}
+                {% set currencies = sylius_channels_currencies() %}
+                {% for channelPricing in product.variants.first.channelPricings %}
+                    {% set channel = channelPricing.channelCode|sylius_channel %}
+                    <tr>
+                        <td>{{ channel.name }}</td>
+                        {% set channelCode = channel.code %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig]: I'm not sure if this variable is needed, as it is used in 2 places and similar to `channel.code`

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig [hunk 7]
```diff
@@ -1,31 +1,29 @@
...
 
-<div class="ui segment">
-    <div class="ui small header">
-        {{ 'sylius.ui.pricing'|trans }}
-    </div>
-    <table class="ui very basic celled table">
-        <tbody>
+<h4 class="ui top attached large header">{{ 'sylius.ui.pricing'|trans }}</h4>
+<div class="ui attached segment">
+
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig]: Unnecessary blank line

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig [hunk 8]
```diff
@@ -1,31 +1,29 @@
...
+<table class="ui very basic celled table">
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig]: Wrong indentation

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig [hunk 9]
```diff
@@ -1,31 +1,29 @@
...
-        </tbody>
-    </table>
+            {% set currencies = sylius_channels_currencies() %}
+            {% for channelPricing in variant.channelPricings %}
+                {% set channel = channelPricing.channelCode|sylius_channel %}
+                <tr class="pricing">
+                    <td>{{ channel.name }}</td>
+                    {% set channelCode = channel.code %}
+                    <td>{{ money.format(channelPricing.price, currencies[channelCode]) }}</td>
+                    {% if channelPricing.originalPrice != null %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig]: `if` could be in `<td></td>`

File: src/Sylius/Bundle/AdminBundle/Twig/ChannelNameExtension.php [hunk 10]
```diff
@@ -30,7 +30,7 @@ public function __construct(ChannelRepositoryInterface $channelRepository)
     public function getFilters(): array
     {
         return [
-            new TwigFilter('sylius_channel_name', [$this, 'getChannelNameByCode']),
+            new TwigFilter('sylius_channel', [$this, 'getChannelNameByCode']),
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Twig/ChannelNameExtension.php]: Should be reverted

File: src/Sylius/Behat/Element/Product/ShowPage/VariantsElement.php [hunk 11]
```diff
@@ -62,7 +64,7 @@ private function hasProductWithGivenNameCodePriceAndCurrentStock(
         if (
             $variant->find('css', '.content .variant-name')->getText() === $name &&
             $variant->find('css', '.content .variant-code')->getText() === $code &&
-            $variantContent->find('css', 'tr.pricing:contains("WEB-US") td:nth-child(2)')->getText() === $price &&
+            $variantContent->find('css',sprintf('tr.pricing:contains("%s") td:nth-child(2)', $channelName) )->getText() === $price &&
```
Comment (GSadee) [inline src/Sylius/Behat/Element/Product/ShowPage/VariantsElement.php]:
```suggestion
            $variantContent->find('css', sprintf('tr.pricing:contains("%s") td:nth-child(2)', $channelName) )->getText() === $price &&
```

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig [hunk 12]
```diff
@@ -11,19 +11,19 @@
...
-                    {% endif %}
-                </tr>
-            {% endfor %}
+                {% set currencies = sylius_channels_currencies() %}
+                {% for channelPricing in product.variants.first.channelPricings %}
+                    {% set channel = channelPricing.channelCode|sylius_channel %}
+                    <tr>
+                        <td>{{ channel.name }}</td>
+                        <td>{{ money.format(channelPricing.price, currencies[channel.code]) }}</td>
+                        {% if channelPricing.originalPrice != null %}
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig]: As below, `if` could be in `<td>`

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig [hunk 13]
```diff
@@ -11,19 +11,21 @@
                 </tr>
             </thead>
             <tbody>
-            {% for channelPricing in product.variants.first.channelPricings %}
-                <tr>
-                    <td class="five wide gray text">
-                        <strong>{{ channelPricing.channelCode|sylius_channel_name }} </strong>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig]: Let's deprecate the usage of `sylius_channel_name` as well in this PR

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig [hunk 14]
```diff
@@ -1,31 +1,29 @@
...
+                        <td>{{ money.format(channelPricing.price, currencies[channel.code]) }}</td>
+                        <td>
+                            {% if channelPricing.originalPrice != null %}
+                                {{ money.format(channelPricing.originalPrice, currencies[channel.code]) }}
+                            {% else %}
+                                <span class="gray text">-</span>
+                            {% endif %}
+                        </td>
+                    </tr>
+                {% endfor %}
```
Comment (Zales0123) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig]: The table in this file and in `src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_pricing.html.twig` is almost the same - maybe we should extract it to some common template? 🤔 We would need to pass only a for loop array as its data. Of course, we can do it in the separate PR, but I fear it would vanish in the doom of abyss of other issues 😄  🕳

Comment (AdamKasp) [reply src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig]: I  opened new PR with fixes of your comment :) https://github.com/Sylius/Sylius/pull/10772

**Comment by Zales0123** [comment]
I would close this PR as one scenario is not passing due to the bug, that is fixed in https://github.com/Sylius/Sylius/pull/10772 💃 It's small enough to merge all at once 🖖

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig [hunk 15]
```diff
@@ -0,0 +1,29 @@
+{% import "@SyliusAdmin/Common/Macro/money.html.twig" as money %}
+
+<h4 class="ui top attached large header">{{ 'sylius.ui.pricing'|trans }}</h4>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_variantContentPricing.html.twig]: Is this file used anywhere? I cannot find usage. Also, won't it be possible to pass always a variant to this template, so we won't need to use two duplicated implementations in other files (one for simple product and another one for more separate variants)
